### ISSUE TITLE
Feature Request: DocC Support

### ISSUE THREAD

**Issue by brevansio** [yonaskolb/XcodeGen#1089]
I know it's early, but I figure this can be used for tracking. DocC seems to be a simple folder with the .docc extension appended to the name. I would be pretty helpful if XcodeGen could detect that.

Currently I have found that running XcodeGen will revert .docc back to a normal folder, so the compiler won't see it as documentation next time.

I'll try to take a look, and see if I can help at all.

**Comment by yonaskolb** [comment]
Should be an easy change here https://github.com/yonaskolb/XcodeGen/blob/master/Sources/ProjectSpec/FileType.swift#L113

**Comment by yonaskolb** [comment]
Would welcome a PR 😄

**Comment by 0xDmitry** [comment]
Hi! I have same problem with Xcode 13.1 and 13.3.1. XcodeGen version is 2.28.0. Can you recommend something?

**Comment by artyom-stv** [comment]
Hi!
The last question is outdated, but I'll leave a comment because this may be helpful to others.
I had a similar problem. It looks like XcodeGen 2.29.0 doesn't add the `.docc` documentation to the "Compile Sources" build phase automatically, and thus the documentation doesn't show up after running "Build Documentation".
This can be fixed by adding `buildPhase: sources` to the corresponding `sources` item  in YML project spec:
```
targets:
  MyTarget:
    ...
    sources:
      - path: Documentation # The directory which contains the `.docc` subdirectory
        buildPhase: sources
      - ...
```

**Comment by freddi-kit** [comment]
Yes https://github.com/yonaskolb/XcodeGen/pull/1202 the fix was it and it is merged into 2.29.0
